src/Token/LcobucciFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key;
+
+if (!class_exists(Key\InMemory::class) && !class_exists(Builder::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Mercure\Token\LcobucciFactory" as the "lcobucci/jwt" package is not installed. Try running "composer require lcobucci/jwt".');
+}
+
+final class LcobucciFactory implements TokenFactoryInterface
+{
+    private $secret;
+
+    public function __construct(string $secret)
+    {
+        $this->secret = $secret;
+    }
+
+    public function create(array $publish, array $subscribe): string
+    {
+        if (class_exists(Key\InMemory::class)) {
+            return ($configuration = Configuration::forSymmetricSigner(
+                new Sha256(),
```

Comment:
This should be configurable, the Mercure hub supports other signing methods.

Reply:
i have added a second argument to the constructor `$algorithm`, this currently supports: hmac.sha256, hmac.sha384, hmac.sha512, ecdsa.sha256, ecdsa.sha384, ecdsa.sha512, rsa.sha256, rsa.sha384, rsa.sha512 ( all sign algorithms provided by lcobucci/jwt ), does mercure support all of them?